src/Event/RouterGenerateEvent.php

```diff
@@ -24,7 +23,7 @@
  *
  * @see \Symfony\Component\Routing\Generator\UrlGeneratorInterface::generate()
  */
-class RouterGenerateEvent extends Event
+class RouterGenerateEvent extends BcEvent
```

Comment:
instead of a new class and file, we could use [class_alias](https://www.php.net/manual/de/function.class-alias.php) before the class declaration, wdyt?

```
if (is_subclass_of(EventDispatcherInterface::class, ContractsEventDispatcherInterface::class)) {
    \class_alias(ContractsEvent::class, 'Event');
} else {
    \class_alias(LegacyEvent::class, 'Event');
}
```

Reply:
I don't think this works because of the 2 event classes, I get this error if I add this to both events

```
Cannot declare class Symfony\Cmf\Component\Routing\Event\Event, because the name is already in use
```

Reply:
hm, there should be no class Event (only Events) in that namespace. but we could name the thing BaseEvent maybe?

Reply:
ah crap, i think now i understood what you say. yeah, we have two events that need this. they would need to check if the other already created the alias - this is becoming a mess, lets keep your solution.